src/wallet/api/wallet.cpp

```diff
@@ -994,6 +994,26 @@ uint64_t WalletImpl::unlockedBalance(uint32_t accountIndex) const
     return m_wallet->unlocked_balance(accountIndex, false);
 }
 
+std::vector<std::pair<std::string, uint32_t>>* WalletImpl::listCurrentStakes() const
+{
+    std::vector<std::pair<std::string, uint32_t>>* stakes = new std::vector<std::pair<std::string, uint32_t>>;
+
+    auto response = m_wallet->list_current_stakes();
+
+    for (rpc::GET_SERVICE_NODES::response::entry const &node_info : response)
+    {
+      for (const auto& contributor : node_info.contributors)
+      {
+        for (size_t i = 0; i < contributor.locked_contributions.size(); ++i)
+        {
+          stakes->push_back(std::make_pair(node_info.service_node_pubkey, contributor.amount));
```

Comment:
`i` is not used here.  This RPC command can return something like this (for someone who contributed multiple times):
```json
      "contributors": [
        {
          "address": "L8UxRW3QS9XLoagM324wwhGBqdYSqLx9rRcJpzDJwEcvWDocqf7WXtshJTSQ7FZ2JyYw2zUzKRbhhXJQuCPfX4YXN1Rn3U2",
          "amount": 5707000000000,
          "locked_contributions": [
            {
              "amount": 3750000000000,
              "key_image": "489a8f0bf2298d1f3a8f38bbb87ca1d53a654b92a130fe80507143ea89901082",
              "key_image_pub_key": "4a7655196dc4bc0dad5c12f3ae9afd7f953b95c8a221e7f964d8168cbc527314"
            },
            {
              "amount": 1957000000000,
              "key_image": "c9889371581c5ff2fb976c3262aec7747f57fcc00a699196975765e2a1dafd61",
              "key_image_pub_key": "35ccfd3d5da602b9c013e4f84a27547e2ba7ed756fcb4fc386d794d800be6b25"
            }
          ],
          "reserved": 5707000000000
        },
```
It probably doesn't matter that we have multiple stakes here in the same SN, so we can just drop this inner for loop and just use the outer amount value (once) for each contributor.